### PR TITLE
add travis configurations

### LINKED ISSUE #11: Add CI support
- [ ] travis support
- [ ] appveyor support

### PR DESCRIPTION
**Pull request by Dylan-DPC**
Closes #11 

I have copied this from another rrepo and made the necessary changes. Hope it works. Can add a rustfmt run on latest nightly and clippy (on stable) if needed.

### CODE REVIEW

**Comment by Dylan-DPC** [comment]
Don't merge this yet, waiting for CI result on fork.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+language: rust
+rust:
+- stable
+- beta
+- nightly
+sudo: false
+install:
+- wget https://github.com/xiph/speexdsp/releases/download/SpeexDSP-1.2rc3.tar.gz
```
Comment (lu-zero) [inline .travis.yml]: Would be better pulling from git in this case

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+- tar xvfz SpeexDSP-1.2rc3.tar.gz
+- cd SpeexDSP-1.2rc3 && ./configure --prefix=$HOME/installed_speedxsp && make && make install &&
+  cd ..
+- export PKG_CONFIG_PATH=$HOME/installed_speedxsp/lib/pkgconfig:$PKG_CONFIG_PATH
+- export LD_LIBRARY_PATH=$HOME/installed_speedxsp/lib:$LD_LIBRARY_PATH
+script:
+- cargo build --verbose
```
Comment (lu-zero) [inline .travis.yml]: Without `--features sys` it just builds the rust transpiled version. Maybe it should test both?

File: .travis.yml (hunk 1 above)
Comment (Dylan-DPC) [reply .travis.yml]: you mean cloning the repo? I'm okay with that but a tar might have been faster

File: .travis.yml (hunk 2 above)
Comment (Dylan-DPC) [reply .travis.yml]: Yes sure, thought they were no features but just checked. Will make the changes

File: .travis.yml (hunk 1 above)
Comment (lu-zero) [reply .travis.yml]: The release code is pretty old sadly ^^;

File: .travis.yml (hunk 2 above)
Comment (lu-zero) [reply .travis.yml]: We could just use the default and avoid the whole speexdsp C library alternatively :)

Comment (Dylan-DPC) [reply .travis.yml]: What do you mean by "avoid the whole library"? :stuck_out_tongue:

Comment (lu-zero) [reply .travis.yml]: `libspeexdsp` is not needed by default, the `sys` feature would pull it in optionally.

File: .travis.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ rust:
 sudo: false
 install:
 - git clone https://github.com/xiph/speexdsp.git
-- cd speexdsp && ./configure --prefix=$HOME/installed_speexdsp && make && make install &&
+- cd speexdsp && make && make install &&
```
Comment (lu-zero) [inline .travis.yml]: You should run `autogen.sh` and then do the `./configure && make` part probably...

(There is a reason why I'd rather go with the first minimal set in :))

Comment (Dylan-DPC) [reply .travis.yml]: Yah will try that. Last attempt probably, else will just test without sys for first run

**Comment by Dylan-DPC** [comment]
@lu-zero is this crate intended to run only on nightly?

**Comment by lu-zero** [comment]
Right now I hadn't had time to fix the c2rust output so for a while we'll be bound to nightly.

**Comment by Dylan-DPC** [comment]
Travis is all set up now. some issue with your tests failing which you can resolve separately.

**Comment by lu-zero** [comment]
Thank you!

**Comment by lu-zero** [comment]
The test seems to pass here...

**Comment by lu-zero** [comment]
Found the problem, now everything seems working fine :)

**Comment by lu-zero** [comment]
Actually now it works locally but fails on travis... There is more to investigate :)

**Pull request merged**